yargs.js

```diff
@@ -772,6 +772,14 @@ function Yargs (processArgs, cwd, parentRequire) {
   }
   self.getStrict = () => strict
 
+  let mergeExtends = false
+  self.mergeExtends = function (enabled) {
```

Comment:
rather than adding a new method `mergeExtends` let's add this as a new `. parserConfiguration`, I think a good name for the configuration setting would be: `deepMergeConfig`, so you'd do this:

```
yargs.parserConfig({'deep-merge-config': true})
```

Reply:
Let me fix that. I just added an entry to https://github.com/yargs/yargs/blob/master/docs/advanced.md#customizing-yargs-parser but let me know if I should do something more.  Really looking forward to this if it gets merge, so I can release my depending projects :)